### PR TITLE
fix(htmlhint): `ruleset` value can be `boolean` or `string`

### PR DESCRIPTION
**Pull request by alan-agius4**
Please fill in this template.

- [X] Make your PR against the `master` branch.
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `tsc` without errors.
- [X] Run `npm run lint package-name` if a `tslint.json` is present.

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [X] Increase the version number in the header if appropriate.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

Changes as per https://github.com/yaniswang/HTMLHint/wiki/space-tab-mixed-disabled

### CODE REVIEW

**Comment by alan-agius4** [comment]
:thumbsup:

File: htmlhint/index.d.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export interface Rule {
 }
 
 export interface RuleSet {
-    [id: string]: boolean;
+    [id: string]: boolean | string;
```
Comment (paulvanbrenk) [inline htmlhint/index.d.ts]: you can make this a string type:

```typescript
type whiteSpaceRule = "tab" | "space" | "mixed" | "disabled";
[id: string] : boolean | whiteSpaceRule;
```

**Comment by paulvanbrenk** [comment]
I had a suggestion, let me know if you don't agree. And I'll merge it regardless.

**Comment by alan-agius4** [comment]
Ill check a bit cause I might be a bit of an overkill to do 3 string unions
considering there are more string rules.

http://htmlhint.com/

I am quite into the idea of having everything strongly typed though.

**Comment by alan-agius4** [comment]
Also, I think `type` namings should be Pascal case similar to Interfaces
ex: `type WhiteSpaceRule = "tab" | "space" | "mixed" | "disabled";`

File: htmlhint/index.d.ts (hunk 1 above)
Comment (alan-agius4) [inline htmlhint/index.d.ts]: Also, I think `type` namings should be Pascal case similar to Interfaces
ex: `type WhiteSpaceRule = "tab" | "space" | "mixed" | "disabled";`

**Comment by paulvanbrenk** [comment]
@alan-agius4 let me know what you want to do, I'm good either way

**Comment by alan-agius4** [comment]
I think i'll leave it as is for the time being.

**Comment by alan-agius4** [comment]
Thank you!

**Pull request merged**